game/src/core/components/DrawComponent.java

```diff
@@ -66,19 +66,26 @@ public class DrawComponent extends Component {
      * @throws IOException if the given path does not exist
      * @see Animation
      */
-    public DrawComponent(Entity entity, String path) throws IOException {
+    public DrawComponent(final Entity entity, final String path) throws IOException {
         super(entity);
         // fetch available animations
         ClassLoader classLoader = getClass().getClassLoader();
-        File directory = new File(classLoader.getResource(path).getFile());
-        if (!directory.exists() || !directory.isDirectory()) {
-            throw new FileNotFoundException("Path " + path + " not found.");
+        File directory;
+        try {
+            directory = new File(classLoader.getResource(path).getFile());
+            animationMap =
+                    Arrays.stream(directory.listFiles())
+                            .filter(File::isDirectory)
+                            .collect(Collectors.toMap(File::getName, Animation::of));
+            currentAnimation(CoreAnimations.IDLE_LEFT);
+        } catch (NullPointerException np) {
+            entity.removeComponent(DrawComponent.class);
```

Comment:
@AMatutat Das solltest Du auf alle Fälle dokumentieren (kurzer Kommentar davor/dahinter). Das versteht sonst in zwei Wochen niemand mehr.

Reply:
done